tests/introspection/tests.py

```diff
@@ -189,6 +184,15 @@ def test_get_indexes_multicol(self):
         self.assertNotIn('first_name', indexes)
         self.assertIn('id', indexes)
 
+    @skipUnlessDBFeature('can_introspect_index_type')
+    def test_get_constraints_indexes_types(self):
+        with connection.cursor() as cursor:
+            constraints = connection.introspection.get_constraints(cursor, Article._meta.db_table)
+        for key, val in constraints.items():
+            if val['index'] and not (val['primary_key'] or val['unique']):
```

Comment:
Okay, I tried doing that but realised that the name is longer than 30 chars so that makes it database dependent (the name would be different for Oracle) so I had to put a vendor check over there as well. Please have a look.